dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowAction.java

```diff
@@ -39,14 +40,49 @@
 	private String icon;
 	private boolean roleHierarchyForAssign;
 	private boolean requiresCheckout;
-	private String  requiresCheckoutOption;
 	private boolean assignable;
 	private boolean commentable;
 	private int order;
+	private Set<WorkflowStatus> showOn = Collections.emptySet();
 
 	public WorkflowAction() {
 	}
 
+	/**
+	 * True if the action should be show on locked status.
+	 * @return boolean
+	 */
+	public boolean shouldShowOnLock () {
+		return this.showOn.contains(WorkflowStatus.LOCKED);
+	}
+
+	/**
+	 * True if the action should be show on unlocked status.
+	 * @return boolean
+	 */
+	public boolean shouldShowOnUnlock () {
+		return this.showOn.contains(WorkflowStatus.UNLOCKED);
+	}
+
+	/**
+	 * Returns the set of the status to show the action.
+	 * @return Set of {@link WorkflowStatus}
+	 */
+	public Set<WorkflowStatus> getShowOn() {
+		return showOn;
+	}
+
+	/**
+	 * Set the set set of the status to show the action.
+	 * @param showOn {@link Set} of {@link WorkflowStatus}
+	 */
+	public void setShowOn(final Set<WorkflowStatus> showOn) {
+		if (null != showOn) {
+			this.showOn = showOn;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "showOn". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)